### PR TITLE
 [Migrate to Gradle 7] | Change compile of internal deps

### PR DESCRIPTION
**Pull request by sudiksha27**
**Goals (and why)**:
On the way to migrating AtlasDb to Gradle 7

### CODE REVIEW

**Comment by carterkozak** [comment]
There's definitely some risk moving from `compile` to `api`, but I expect library bumps in our large internal friend will validate things quickly enough.
I might have opted to swap `compile` -> `api` (since this is guaranteed to be safe) then separately worked to swap `api` down to `implementation` where reasonable, but either way works.

File: lock-impl/build.gradle [hunk 1]
```diff
@@ -1,8 +1,8 @@
 apply from: "../gradle/shared.gradle"
 
 dependencies {
-  compile project(":lock-api")
-  compile project(":atlasdb-commons")
+  implementation project(":lock-api")
```
Comment (jeremyk-91) [inline lock-impl/build.gradle:4]: this one also seems like it should be `api`?

**Review by jeremyk-91 — COMMENTED** [review]
Thanks a lot for doing this! This looks pretty solid, and I've noted that we have green builds of both large internal product and atlasdb proxy, so I'm pretty confident we'll be able to integrate this feasibly.

I think there are a few deps where the visibility doesn't really match what I'd expect. Generally, think of `api` as a public dependency and `implementation` as a private one - this does mean that for e.g. `lock-impl` I think `lock-api` should be public, since you can't really use `lock-impl` if you don't also have `lock-api` present. Conversely, for `commons-api` it isn't clear to me which class we need from `atlasdb-commons` as part of the public api. We can get some time tomorrow to discuss.

Especially for the ones that I think should be `implementation` instead of `api`, though, it's definitely fine to just leave a comment to check it as opposed to changing it now, realising that this is already a massive improvement over either not migrating, or the "everything api" approach that I'd have tried first.

File: atlasdb-config/build.gradle [hunk 2]
```diff
@@ -2,12 +2,12 @@
...
 
 dependencies {
-    compile project(':atlasdb-api')
-    compile project(':atlasdb-conjure')
-    compile project(':atlasdb-impl-shared')
-    compile project(':leader-election-impl')
-    compile project(':lock-impl')
-    compile project(':timestamp-client')
+    api project(':atlasdb-api')
+    api project(':timestamp-client')
```
[the remark this answers is not in the document]
Comment (sudiksha27) [reply atlasdb-config/build.gradle:6]: Most likely not - cannot remember why I did it this way but will leave a comment.

File: commons-api/build.gradle [hunk 3]
```diff
@@ -1,9 +1,9 @@
 apply from: "../gradle/shared.gradle"
 
 dependencies {
-  compile project(':atlasdb-commons')
-  compile project(':commons-annotations')
+  api project(':atlasdb-commons')
```
[the remark this answers is not in the document]
Comment (sudiksha27) [reply commons-api/build.gradle:4]: There is JMXUtils and Pair (belong to `atlasdb-commons`) which are required by `commons-api`. But probably `atlasdb-commons` is not required to be exposed as public api here.

File: gradle/shared.gradle [hunk 4]
```diff
@@ -92,7 +92,7 @@ tasks.withType(Test) {
 }
 
 task allCompileDeps(type: DependencyReportTask) {
-    configurations = [project.configurations.getByName("compile")]
+    configurations = [project.configurations.getByName("api")]
```
[the remark this answers is not in the document]
Comment (sudiksha27) [reply gradle/shared.gradle:95]: Gonna delete the method and script

File: timelock-server/build.gradle [hunk 5]
```diff
@@ -40,7 +40,7 @@ check.dependsOn suiteTest
 check.dependsOn stressTest
 
 dependencies {
-    compile project(":timelock-agent")
+    api project(":timelock-agent")
```
[the remark this answers is not in the document]
Comment (sudiksha27) [reply timelock-server/build.gradle:43]: we will bump it down

**Review by jeremyk-91 — APPROVED** [review]
👍

**Pull request merged**